scipy/optimize/tests/test_zeros.py

```diff
@@ -809,6 +809,30 @@ def f(x):
         assert res[1].function_calls == f.calls
 
 
+def test_gh_14486_converged_false():
+    """Test that zero slope with secant method results in a converged=False"""
+    def lhs(x):
+        return x * np.exp(-x*x) - 0.07
+
+    with pytest.warns(RuntimeWarning, match='Tolerance of'):
+        res = root_scalar(lhs, method='secant', x0=-0.15, x1=1.0)
+    assert not res.converged
+    assert res.flag == 'convergence error'
+
+    with pytest.warns(RuntimeWarning, match='Tolerance of'):
+        res = newton(lhs, x0=-0.15, x1=1.0, disp=False, full_output=True)[1]
+    assert not res.converged
+    assert res.flag == 'convergence error'
+
+    # This test case doesn't fail to converge for the vectorized version of
```

Comment:
There are two "this" in the statement. Does one refer to the function `lhs` above perhaps?

Reply:
Thanks @pvanmulbregt. Yeah, that was pretty confusing. How does the suggested change look?